src/Boids.hs

```diff
@@ -51,9 +54,14 @@ guideBoidToPoint :: Point -> Boid -> Boid
 guideBoidToPoint guidePoint boid = guideBoidToVector direction boid
     where direction = fromPoints (boidPosition boid) guidePoint
 
+
+isWithinView :: Boid -> Boid -> Bool
+isWithinView boid1 boid2 = angleVV (unitVectorAtAngle $ boidHeading boid1) (fromPoints (boidPosition boid1) (boidPosition boid2)) <= viewAngle
+
 getNearBoids :: Boid -> Float -> [Boid] -> [Boid]
-getNearBoids boid boidDistance boids = filter isTooClose boids
-    where isTooClose boid' = distance (boidPosition boid) (boidPosition boid') <= boidDistance
+getNearBoids pivotBoid nearDistance boids = filter isVisible boids
```

Comment:
I think "proximity" is a better name than "near" distance, mainly because it's an actual word with clear meaning while `nearDistance` requires some context.